src/test/java/com/kodilla/ecommercee/repositoryDao/CartRepositoryTestSuit.java

```diff
@@ -0,0 +1,184 @@
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        //When
+        List<Cart> listOfCarts = cartRepository.findAll();
+        //Then
+        Assert.assertEquals(3,listOfCarts.size());
+        Assert.assertNotEquals(null,cartRepository.findById(cart1.getId()));
+        Assert.assertNotEquals(null,cartRepository.findById(cart2.getId()));
+        Assert.assertNotEquals(null,cartRepository.findById(cart3.getId()));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoSaveAndFindById(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long IdCart1 = cart1.getId();
+        //When
+        Optional<Cart> cartFound = cartRepository.findById(IdCart1);
+        //Then
+        Assert.assertEquals(cart1.getId(),cartFound.get().getId());
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoDeleteById(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long cart2Id = cart2.getId();
+        //When
+        cartRepository.deleteById(cart2Id);
+        List<Cart> cartList = cartRepository.findAll();
+        //Then
+        Assert.assertFalse(cartList.contains(cart2));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+
+    @Transactional
+    @Test
+    public void testCartOneOnOneWithOrder(){
+        //Given
+        Order order = new Order();
+        cart1.setOrder(order);
+        //When
+        cartRepository.save(cart1);
+        Assert.assertNotEquals(0,cartRepository.findById(cart1.getOrder().getOrderId()));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartManyToOneUser(){
+        //Given
+        user1 = new Users();
+        user1.getCarts().add(cart1);
+        user1.getCarts().add(cart2);
+        //When
```

Comment:
W sekcji //When powinno być zapisanie cart1 i cart2 do repozytorium (tak jak robisz to w pozostałych testach, tutaj przeoczyłeś)

Reply:
Ola nie przeoczylem tylko nie widze sensu po co zapisywac cos co nie jest sprawdzane w danym momencie, chyba ze Ty to widzisz inaczej to prosze wytlumacz mi, dzieki

Reply:
Cała idea tych testów polega na sprawdzeniu czy poprawnie działa repozytorium. Dopóki nie wykonasz save danego obiektu poprzez repozytorium to nie jest on w bazie danych. W tym teście, w formie którą masz teraz,  tak jakby sprawdzasz czy działa getter do usera... a nie o to chodzi. Powinieneś sprawdzać czy user1 podpięty w relacji do cart1 i cart2 zapisze się w bazie danych w momencie gdy do bazy danych zapiszesz cart1/cart2. O to chodzi. I w asercji powinno być sprawdzenie assertNotEquals(zeroLong, user1.getId()) i jeszcze asercje sprawdzające istnienie id dla cart1 i cart2

Reply:
Ok Ola dzieki zastosowalem sie do tego zrobilem analogicznie jak Ty w testachProduct , dzieki